src/core/webview/ClineProvider.ts

```diff
@@ -241,42 +244,51 @@ export class ClineProvider implements vscode.WebviewViewProvider {
 	public async initClineWithTask(task?: string, images?: string[]) {
 		await this.clearTask()
 		const {
-			apiConfiguration,
-			customInstructions,
-			diffEnabled,
-			fuzzyMatchThreshold
-		} = await this.getState()
-
-		this.cline = new Cline(
-			this,
 			apiConfiguration,
 			customInstructions,
 			diffEnabled,
 			fuzzyMatchThreshold,
-			task,
-			images
-		)
+			slackConfig
+		} = await this.getState()
+			this.cline = new Cline(
+				this,
+				apiConfiguration,
+				customInstructions,
+				diffEnabled,
+				fuzzyMatchThreshold,
+				task,
+				images,
+				undefined,
+				{
+					enabled: slackConfig?.enabled ?? false,
+					webhookUrl: slackConfig?.webhookUrl ?? ""
+				}
```

Comment:
Should we just pass in the slackConfig here? And should it go before the historyItem?